Assets/_BForBoss/_Character/Scripts/FirstPersonPlayer.Animator.cs

```diff
@@ -0,0 +1,38 @@
+using ECM2.Characters;
+using UnityEngine;
+
+namespace BForBoss
+{
+    public partial class FirstPersonPlayer : FirstPersonCharacter
+    {
+        #region ANIMATOR_FIELDS
+        private static readonly int ForwardParamId = Animator.StringToHash("Forward");
+        private static readonly int TurnParamId = Animator.StringToHash("Turn");
+        private static readonly int GroundParamId = Animator.StringToHash("OnGround");
+        private static readonly int CrouchParamId = Animator.StringToHash("Crouch");
+        private static readonly int JumpParamId = Animator.StringToHash("Jump");
+        #endregion
+
+        protected override void Animate()
+        {
+            if (!IsThirdPerson) return;
+
+            // Compute input move direction vector in local space
+            Vector3 move = rootPivot.InverseTransformDirection(GetMovementDirection());
+
+            float forwardAmount = useRootMotion ? move.z : Mathf.InverseLerp(0.0f, GetMaxSpeed(), GetSpeed());
+
+            animator.SetFloat(ForwardParamId, forwardAmount, 0.1f, Time.deltaTime);
```

Comment:
i know it's work in progress, so idk if u wanna do it now, but it's better to separate each `animator.Set` to its own method, so it's easier to read what each thing is doing.

Reply:
Fixed, I'll clean it up further once I had in slide logic